### ISSUE TITLE
LGraphCanvas.onSelectionChange callback called twice

### ISSUE THREAD

**Issue by altarfinch** [jagenjo/litegraph.js#147]
When switching selection from one node to the other, we get two events.

For example:
1. Node 0 is selected
2. We click on node 1
3. We receive one event saying selection is now []
4. We receive another event saying selection is now [1]

Is it the expected behaviour ?
If you like I can modify it, that's an easy fix.

**Comment by jagenjo** [comment]
yes, because first you receive the deselection, and after that the selection of the new item

**Comment by altarfinch** [comment]
I understand that these are valid intermediate steps.

However, wouldn't it be more efficient to only trigger one event per action? Then it would be up to the developper to compare previous and new list of nodes and do the minimum modifications. 

Currently, it is not possible.

If one select many nodes : [0,1,2,3,4,5,6], and then one more : [0,1,2,3,4,5,6,7].

One will get [] and then [0,1,2,3,4,5,6,7], so potentially, one will have to create new objects which one could have reused.

**Comment by xuekai-china** [comment]
> yes, because first you receive the deselection, and after that the selection of the new item

LGraphCanvas.onSelectionChange not support typescript 

![1609991412166](https://user-images.githubusercontent.com/49433924/103849094-d9aeb780-50de-11eb-838a-471a01e2f831.jpg)
